product_pricelist_supplierinfo_coefficient/views/product_pricelist_item_views.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="product_pricelist_item_form_view_inherit" model="ir.ui.view">
+        <field name="name">product.pricelist.item.form.inherit</field>
+        <field name="model">product.pricelist.item</field>
+        <field name="inherit_id" ref="product.product_pricelist_item_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='base']" position="after">
+                <field name="coefficient" invisible="base != 'supplierinfo'"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="product_pricelist_item_tree_view_inherit" model="ir.ui.view">
+        <field name="name">product.pricelist.item.tree.inherit</field>
+        <field name="model">product.pricelist.item</field>
+        <field name="inherit_id" ref="product.product_pricelist_item_tree_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='price']" position="before">
+                <field name="coefficient"/>
```

Comment:
Suggested code (column is still visible, but inner cell data is not shown when `coefficient` has no sense for a certain record, so a lot of unnecessary shown `0.00` values is avoided), not tested:
```suggestion
                <field name="base" column_invisible="1"/>
                <field name="coefficient" invisible="base != 'supplier_info'"/>
```